### ISSUE TITLE
Application asks for extremely wide permissions on macOS, without explanation

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [Yubico/yubioath-flutter#912]
Without giving any popup explaining itself, upon launch, the application results in this prompt on macOS:

<img width="573" alt="image" src="https://user-images.githubusercontent.com/200/212782994-7f19c6b6-973c-4f9e-9375-e4babcd3043e.png">

This is one of the most dangerous permissions available on macOS, and should _absolutely_ not be used lightly. At the very least, requesting this permission (for … whatever it's used for?) should be 1. opt-in in the settings, and 2. include a pop-up describing _why_ you're going to ask for that permission, before actually asking for it.

If it's helpful, here's an example of a well-behaved app discovering at launch that it doesn't have the accessibility permissions it wants:

<img width="912" alt="image" src="https://user-images.githubusercontent.com/200/212783290-e0b4e951-b61d-448c-8f47-fb8afc9aa098.png">

**Comment by dainnilsson** [comment]
I agree that that app could do a (much) better job in itself to explain the permissions needed, we'll look into improving this.

To answer the immediate question: The app requires this permission to be able to access the YubiKey over one of the USB interfaces that the YubiKey exposes, specifically the interface that tells the host that it is a USB keyboard. This interface is used for the Yubico OTP protocol, but also for certain configuration of the YubiKey itself.

Here's a short article on why the YubiKey appears as a keyboard: https://support.yubico.com/hc/en-us/articles/360013790279
